hail/python/hail/matrixtable.py

```diff
@@ -539,6 +539,138 @@ class MatrixTable(ExprContainer):
+        cols: Optional[Dict[str, Iterable[Any]]] = None,
+        entries: Optional[Dict[str, Iterable[Iterable[Any]]]] = None
+    ) -> 'MatrixTable':
+        """Create a `MatrixTable` from its component parts.
+
+        Example
+        -------
+        >>> mt = hl.MatrixTable.from_parts(
+        ...     globals={'hello':'world'},
+        ...     rows={'foo':[1, 2]},
+        ...     cols={'bar':[3, 4]},
+        ...     entries={'baz':[[1, 2],[3, 4]]}
+        ... )
+        >>> mt.describe()
+        ----------------------------------------
+        Global fields:
+            'hello': str
+        ----------------------------------------
+        Column fields:
+            'col_idx': int32
+            'bar': int32
+        ----------------------------------------
+        Row fields:
+            'row_idx': int32
+            'foo': int32
+        ----------------------------------------
+        Entry fields:
+            'baz': int32
+        ----------------------------------------
+        Column key: None
+        Row key: ['row_idx']
+        ----------------------------------------
+        >>> mt.row.show()
+        +---------+-------+
+        | row_idx |   foo |
+        +---------+-------+
+        |   int32 | int32 |
+        +---------+-------+
+        |       0 |     1 |
+        |       1 |     2 |
+        +---------+-------+
+        >>> mt.col.show()
+        +---------+-------+
+        | col_idx |   bar |
+        +---------+-------+
+        |   int32 | int32 |
+        +---------+-------+
+        |       0 |     3 |
+        |       1 |     4 |
+        +---------+-------+
+        >>> mt.entry.show()
+        +---------+-------------+-------------+
+        | row_idx | <col 0>.baz | <col 1>.baz |
+        +---------+-------------+-------------+
+        |   int32 |       int32 |       int32 |
+        +---------+-------------+-------------+
+        |       0 |           1 |           2 |
+        |       1 |           3 |           4 |
+        +---------+-------------+-------------+
+
+        Parameters
+        ----------
+        globals : :class:`dict` from :class:`str` to :obj:`any`
+            Global fields by name
+
+        rows: :class:`dict` from :class:`str` to :class:`list` of :obj:`any`
+            Row fields by name
+
+        cols: :class:`dict` from :class:`str` to :class:`list` of :obj:`any`
+            Column fields by name
+
+        entries: :class:`dict` from :class:`str` to :class:`list` of :class:`list` of :obj:`any`
+            Matrix entries by name in the form `entry[row_idx][col_idx]`
+
+        Returns
+        -------
+        :class:`.MatrixTable`
+            A MatrixTable assembled from inputs keyed by `row_idx`
+
+        Notes
+        -----
+        - The number of rows and columns specified in `rows` and `cols` for all
+          fields must match the matrix dimensions in `entries`.
+        - This method is intended for educational and testing purposes only.
```

Comment:
I think we should prefer a statement of limitations. Something like: "This method assumes all the annotations are small enough to fit in memory."

Hmm. They clearly fit in RAM though since they're Python objects passed as arguments to this method. Is there something in particular you're trying to hedge against with this Note?

Reply:
I'm trying to say it's the caller's responsibility to make sure matrix dimensions agree. Enforcing dimensions will make remove the need for it